### PR TITLE
[ROCm] replace ROCmLoops.cuh with hipified CUDALoops.cuh

### PR DESCRIPTION
**Pull request by jeffdaily**
The intent of this change was to minimize code differences between CUDA and ROCm while maintaining or improving performance.

Verified new performance using pytorch/benchmarks/operator_benchmark.

```
python -u -m pt.unary_test --tag-filter all --device cuda
python -u -m pt.binary_test --tag-filter all --device cuda
```

On MI200 this improved performance on average 3%.

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Code cleanup :tada: 
Thanks!

File: aten/src/ATen/native/cuda/Loops.cuh [hunk 1]
```diff
@@ -68,17 +68,7 @@ __device__ inline void elementwise_kernel_helper(func_t f, policy_t policy) {
...
-//   https://github.com/pytorch/pytorch/pull/32383
-// Because for some reason trying to enable vectorized
-// memory access introduce regression on ROCm.
-
-#if !defined(USE_ROCM)
-  #include <ATen/native/cuda/CUDALoops.cuh>
-#else
-  #include <ATen/native/cuda/ROCmLoops.cuh>
-#endif
+#include <ATen/native/cuda/CUDALoops.cuh>
```
[the remark this answers is not in the document]
Comment (jeffdaily) [reply aten/src/ATen/native/cuda/Loops.cuh:71]: The regressions we saw last time we tried this were caught by the operator benchmarks.  This time around, I root-caused it to the kernels that require dynamic casting and ported the ROCmLoops version of that code into CUDALoops and #ifdef'd it.  The operator benchmarks now confirm zero regressions.  I'm satisfied.

**Pull request closed without merging**